sktime/dists_kernels/scipy_dist.py

```diff
@@ -0,0 +1,88 @@
+# -*- coding: utf-8 -*-
+"""
+Interface module to scipy.spatial's pairwise distance function cdist
+    exposes parameters as scikit-learn hyper-parameters
+"""
+
+__author__ = ["fkiraly"]
+
+import pandas as pd
+
+from scipy.spatial.distance import cdist
+
+from sktime.dists_kernels._base import BasePairwiseTransformer
+
+
+class ScipyDist(BasePairwiseTransformer):
+    """
+    computes pairwise distances using scipy.spatial.distance.cdist
+        includes Euclidean distance and p-norm (Minkowski) distance
+            note: weighted distances are not supported
+
+    Hyper-parameters:
+        metric: string or function, as in cdist; default = 'euclidean'
+            if string, one of: 'braycurtis', 'canberra', 'chebyshev', 'cityblock',
+                'correlation', 'cosine', 'dice', 'euclidean', 'hamming', 'jaccard',
+                'jensenshannon', 'kulsinski', 'mahalanobis', 'matching', 'minkowski',
+                'rogerstanimoto', 'russellrao', 'seuclidean', 'sokalmichener',
+                'sokalsneath', 'sqeuclidean', 'yule'
+            if function, should have signature 1D-np.array x 1D-np.array -> float
+        p: if metric='minkowski', the "p" in "p-norm", otherwise irrelevant
+        colalign: string, one of 'intersect' (default), 'force-align', 'none'
+            controls column alignment if X, X2 passed in fit are pd.DataFrame
+            columns between X and X2 are aligned via column names
+            if 'intersect', distance is computed on columns occurring both in X and X2,
+                other columns are discarded; column ordering in X2 is copied from X
+            if 'force-align', raises an error if the set of columns in X, X2 differs;
+                column ordering in X2 is copied from X
+            if 'none', X and X2 are passed through unmodified (no columns are aligned)
+                note: this will potentially align "non-matching" columns
+    """
+
+    _tags = {
+        "symmetric": True,  # all the distances are symmetric
```

Comment:
Major:
I am not sure about this. There is high chance you are getting mixed up between symmetry of elementwise transformer and symmetry of pairwise transformer. the distance function is symmetric, but that is different from pairwise transformer being symmetric

* in base class, you say symmetric tag means: `# is the transformer symmetric, i.e., t(x,y)=t(y,x) always`
* However, trivially, all pairwise transformer are not symmetric. Just let len(x) != len(y), and the resulting dist_matrix have different shapes
* maybe symmetric condition should be something like `t(x,y) = transpose(t(y,x))`? (and that definition of aggfunc being symmetric is that aggfunc(M) = aggfunc(transpose(M)) )

I think the only way to properly resolve this is to get out some pen and paper, and actually write out explicitly some definitions and some proofs of relevant claims (main claim being that if various boolean tags/variables are true, then the final panel pairwise transformer is symmetirc)

Reply:
@Lovkush-A, ok, I think I see the point of confusion here.

All transformers in the `dists_kernels` module are meant to be element-wise, but vectorized.

I.e., they all model functions `d: R^d x R^d -> R` (in the tabular case) or functions `d: series x series -> R` (in the panel case.

The machine type implements a vectorized version of this, i.e., `d: R^(nxd) x R^(mxd) -> R^(mxn)` resp. `d: series^n x series^m -> R^(mxn)`, this is precisely also the convention from `scipy`, or from `scikit-learn` (kernels for GP).

Reply:
this is a "won´t fix" since transformers are always vectorized.